### ISSUE TITLE
Create GCE resources in parallel

### ISSUE THREAD

**Issue by madhusudancs** [GoogleCloudPlatform/k8s-multicluster-ingress#43]
There is room for optimization. It can take a few seconds to create certain GCP resources and we could improve the end-to-end lb creation time by creating GCP resources in parallel.

**Comment by madhusudancs** [comment]
cc @nikhiljindal @csbell

**Comment by nikhiljindal** [comment]
We can create health checks for all clusters in parallel.

Many different resources have a dependency.
For ex: backend services cant be created unless health checks are created, same for target proxy and url maps.

Whatever we do here will be relevant for both create and delete.

**Comment by madhusudancs** [comment]
Yeah, whatever can be parallelized should be parallelized. Per-cluster resources are good candidates. Even global resources like firewall for example are candidates. It feels like firewall rule creation is one of the slowest operations and it doesn't depend on anything else.
